client/app/dbaas/logs/detail/streams/logs-streams.service.js

```diff
@@ -247,6 +251,73 @@ class LogsStreamsService {
         };
     }
 
+    /**
+     * extracts graylog URL from stream. Shows error message on UI if no graylog URL is found.
+     *
+     * @param {any} stream
+     * @returns {string} graylog url, if not found empty string
+     * @memberof LogsStreamsService
+     */
+    getStreamGraylogUrl (stream) {
+        const url = this.UrlHelper.findUrl(stream, "GRAYLOG_WEBUI");
+        if (!url) {
+            this.CloudMessage.error({ textHtml: this.$translate.instant("logs_streams_get_graylog_url_error", { stream: stream.info.title }) });
+        }
+        return url;
+    }
+
+    /**
+     * extracts and copies stream token to clipboard.
+     * Shows error message on UI if no no token found or browser does not support clipboard copy.
+     *
+     * @param {any} stream
+     * @memberof LogsStreamsService
+     */
+    copyStreamToken (stream) {
+        const token = this.getStreamToken(stream);
+        if (token) {
+            const error = this.ControllerHelper.copyToClipboard(token);
+            if (error) {
+                this.CloudMessage.error({ textHtml: this.$translate.instant("logs_streams_copy_token_error", {
+                    stream: stream.info.title,
+                    token_value: token
+                }) });
+            } else {
+                this.CloudMessage.success(this.$translate.instant("logs_streams_copy_token_success"));
```

Comment:
<translation id="logs_streams_copy_token_error">Your browser does not support copy to clipboard. Please upgrade browser to latest version.<br/>Here is the token value to enter for stream <strong>{{stream}}</strong>: <strong>{{token_value}}</strong></translation>

I used textHTML because the localized message has HTML elements. Is there any alternative way to show strong text in error and success alerts?

Reply:
I removed textHTML wherever HTML elements are not used in translation. I have retained it in places where <strong> and <br> html elements are used in translation. Hope this is fine. Let me know if you have any better idea? do we show strong text in alerts? is it ok to show?